service/service.py

```diff
@@ -107,19 +108,80 @@ def index():
 
 
 ######################################################################
-# RETRIEVE A PROMOTION
+# Configure Swagger before initializing it
 ######################################################################
-@app.route("/promotions/<int:promotion_id>", methods=["GET"])
-def get_promotions(promotion_id):
+api = Api(app,
+          version='1.0.0',
+          title='Promotions REST API Service',
+          description='This is the Promotions server.',
+          default='promotions',
+          default_label='Promotions service operations',
+          doc='/api-docs',
+          # authorizations=authorizations,
+          prefix='/'
+         )
+
+# Define the model so that the docs reflect what can be sent
+create_model = api.model('Promotion', {
+    'title': fields.String(required=False,
+                          description='The name of the promotion'),
+    'description': fields.String(required=False,
+                              description='The description of the promotion'),
+    'promo_code': fields.String(required=False,
+                              description='The promo code associated with this promotion'),
+    'promo_type': fields.String(required=False,
+                              description='The type of promotion [BOGO | DISCOUNT | FIXED]'),
+    'amount': fields.Integer(required=False,
+                              description='The amount of the promotion based on promo type'),
+    'start_date': fields.DateTime(required=False,
+                              description='The start date of the promotion'),
+    'end_date': fields.DateTime(required=False,
+                              description='The end date of the promotion'),
+    'is_site_wide': fields.Boolean(required=False,
+                                description='Is the promotion site wide?')
+})
+
+promotion_model = api.inherit(
+    'PromotionModel', 
+    create_model,
+    {
+        'id': fields.Integer(readOnly=True,
+                            description='The unique id assigned internally by service'),
+    }
+)
+
+
+######################################################################
+#  PATH: /promotions/{id}
+######################################################################
+@api.route("/promotions/<int:promotion_id>")
+@api.param('promotion_id', 'The Promotion identifier')
+class PromotionResource(Resource):
     """
-    Retrieve a single Promotion
-    This endpoint will return a Promotion based on it's id
+    PromotionResource class
+
+    Allows the retrieval/manipulation of a single promotion
+    GET /pet/{id} - Returns the promotion with the id
```

Comment:
probably want to change these to `promotion`

Reply:
Oops! Thanks for that, fixed it in the latest commit.